_posts/2023-11-20-unlocking-the-secrets-to-ingestion.md

```diff
@@ -0,0 +1,191 @@
+
+![Document Replication](/assets/media/blog-images/2023-11-20-unlocking-the-secrets-to-ingestion/DocumentReplication.svg){: .img-fluid }
+
+With segment replication enabled the first two steps remain the same. Once it gets to the third step is where the magic happens. Instead of sending the original unprocessed document into the replicas we send the processed Lucene segment over the network. Now the nodes that the replicas live on will be free to use more of their compute for ingestion. This comes with a few caveats though. The first is you will be using more bandwidth between nodes as the segments are much larger than the initial documents. It’s important to consider your network topology here. Second, this model is an eventually consistent model as the document updates are not available as readily on replicas. The final, is you will want to increase the refresh interval of the nodes so you do not overwhelm the replica nodes with network traffic. We’ll discuss that more in the next section.
+
+![Segment Replication](assets/media/blog-images/2023-11-20-unlocking-the-secrets-to-ingestion/SegmentReplication.svg){: .img-fluid }
+
+This setting needs to be applied before the index is created. If you have an index you would like to convert to segment replication you can create a new index and use one of OpenSearch’s reindex api’s to get the data in. Here is how you could configure segment replication for an index.
+
+```json
+POST index-name/_settings
+{
+	"index": {
+		"replication.type": "SEGMENT"
+	}
+}
+```
+
+### Refresh Interval
+
+This setting represents amount of time between when a document is received and when it is sent to replicas. This can either be set at a cluster level or on a per-index level. By default OpenSearch uses a 1 second refresh interval but that is probably faster than needed for most people with the observability use case. Below is how you could set that in the index settings. You can change this refresh interval even after the index has already been created.
+
+```json
+POST index-name/_settings
+{
+	"index": {
+		"refresh_interval": "30s"
+	}
+}
+```
+
+As we mentioned in the previous section we increase this delay so that we are sending segments less frequently. They are much larger than the original documents so sending them more quickly could end up saturating the connection of a node.
+
+# Compression Settings
+
+When you are storing large amounts of data it can be handy to use compression to minimize on disk size. We have a few different compression algorithms that you can use but the one we will talk about is Zstandard compression (ZSTD). This compression works well because it is tunable, meaning it lets you prioritize either speed or level of compression. Below is the setting we can use to enable ZSTD compression. Note, the compression level is a default of 3 with 6 being the most compressed and 1 being the least.
```

Comment:
```suggestion
When you are storing large amounts of data it can be handy to use compression to minimize on disk size. We have a few different compression algorithms that you can use. The one we will talk about is Zstandard compression (ZSTD). This compression works well because it's tunable, meaning it lets you prioritize either speed or level of compression. Below is the setting we can use to enable ZSTD compression. Note, the compression level is a default of 3, with 6 being the most compressed and 1 being the least.
```